localstack-core/localstack/services/dynamodb/server.py

```diff
@@ -142,9 +142,28 @@ def jar_path(self) -> str:
     def library_path(self) -> str:
         return f"{dynamodblocal_package.get_installed_dir()}/DynamoDBLocal_lib"
 
+    def _get_java_vm_options(self) -> list[str]:
+        dynamodblocal_installer = dynamodblocal_package.get_installer()
+
+        # Workaround for JVM SIGILL crash on Apple Silicon M4
+        # See https://bugs.openjdk.org/browse/JDK-8345296
+        # To be removed after Java is bumped to 17.0.15+ and 21.0.7+
```

Comment:
Eclipse Temurin haven't released the latest builds yet

https://github.com/adoptium/temurin17-binaries/releases

https://github.com/adoptium/temurin21-binaries/releases